apps/storefront/app/root.tsx

```diff
@@ -31,6 +40,23 @@ export const meta: MetaFunction<typeof loader> = mergeMeta(getCommonMeta, getRoo
 
 export const loader = getRootLoader;
 
+export const shouldRevalidate: ShouldRevalidateFunction = ({
+  actionResult,
+  currentParams,
+  currentUrl,
+  defaultShouldRevalidate,
+  formAction,
+  formData,
+  formEncType,
+  formMethod,
+  nextParams,
+  nextUrl,
+}) => {
+  if (!formMethod || formMethod === 'GET') return false;
```

Comment:
essentially: we only need to re-load this root data if the customer makes any mutation.